include/Nodes/Party.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "Nodes/Node.hpp"
+#include "Nodes/Actor.hpp"
+
+#include <vector>
+#include <memory>
+#include <algorithm>
+
+// Grupa aktorów jako węzeł w drzewie sceny
+class Party : public Node {
```

Comment:
When merged with main branch rewrite this class as a ContainerNode
```suggestion
class Party : public ContainerNode<Actor>
```
You should then be able to use the functions specific to ContainerNode instead